hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -354,12 +350,21 @@ extends Importer:
       val preTrm = term(pre)
       val sym = resolveField(nme, preTrm.symbol, nme)
       Term.SynthSel(preTrm, nme)(sym)
-    case Sel(pre, nme) =>
+    case t @ Sel(pre, nme) =>
       val preTrm = term(pre)
       val sym = resolveField(nme, preTrm.symbol, nme)
-      if inAppPrefix
+      val res = if inAppPrefix
       then Term.SynthSel(preTrm, nme)(sym)
       else Term.Sel(preTrm, nme)(sym)
+      val isGetter = ctx.get(nme.name) match // TODO: create a function for this logic
+        case S(_: Ctx.GetElem) => true
+        case _ => sym.flatMap(_.defn) match
```

Comment:
The code should not have to access `defn` to work. The relevant info could be stored in the symbol itself I think.